sdk/src/test/scala/com/horizen/account/state/WithdrawalMsgProcessorTest.scala

```diff
@@ -167,4 +171,38 @@ class WithdrawalMsgProcessorTest extends JUnitSuite with MockitoSugar with Withd
     returnData = withGas(WithdrawalMsgProcessor.process(msg, mockStateView, _, defaultBlockContext), 10000000)
     assertArrayEquals(WithdrawalRequestsListEncoder.encode(expectedListOfWR), returnData)
   }
+
+  @Test
+  def testRejectSendingPositiveValueToGetListOfWithdrawal(): Unit = {
+    val mockStateView = mock[AccountStateView]
+
+    val msg = getMessage(
+      WithdrawalMsgProcessor.contractAddress,
+      BigInteger.ONE,
+      BytesUtils.fromHexString(WithdrawalMsgProcessor.GetListOfWithdrawalReqsCmdSig)
+    )
+
+    val returnData = intercept[ExecutionRevertedException] {
+      withGas(WithdrawalMsgProcessor.process(msg, mockStateView, _, defaultBlockContext))
+    }
+    assert(returnData.getMessage === "Call value must be zero")
```

Comment:
Sure, but what if the method has multiple parts where the same exception is thrown? How do we know then that the exception was thrown at the exact part we are trying to test?

Reply:
You don't and you shouldn't. You setup a test case with an invalid message and expect execution to revert and that's it. Anything more and you're going from testing the expected behavior to testing how it was implemented.

If this case was important enough it would have its own exception type which you could test for.